apps/web/src/utils/logger.ts

```diff
@@ -20,16 +26,41 @@ class CustomLogger {
     return CustomLogger.instance;
   }
 
-  private log(level: LogLevel, message: string, meta?: Record<string, unknown>) {
+  private createDatadogLog(level: LogLevel, message: string, meta?: Record<string, unknown>) {
+    let traceId: string | undefined;
+    let spanId: string | undefined;
+
+    //TODO: initialice ddTrace through dd-tracer
+    if (ddTrace) {
+      // Access trace information server-side
+      const currentSpan = ddTrace?.scope().active();
+      traceId = currentSpan?.context().toTraceId() ?? undefined;
+      spanId = currentSpan?.context().toSpanId() ?? undefined;
```

Comment:
similarly here ` currentSpan?.context()?.toSpanId()` ? if context() is null or undefined we'll get an error.

Reply:
context() shouldn't be nil, since it is part of the definition of the object itself